packages/python/pyfora/typeConverters/Pandas.py

```diff
@@ -0,0 +1,169 @@
+#   Copyright 2015 Ufora Inc.
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+
+import pyfora.PureImplementationMapping as PureImplementationMapping
+
+import pandas as pd
+
+
+#######################################
+# PurePython implementations:
+#######################################
+
+class _DataframeFactory:
+    def __call__(self, data):
+        return PurePythonDataFrame(data)
+
+
+class PurePythonDataFrame:
+    """
+    TODO `columns` attr is an Index of column names in Pandas.
+    """
+    def __init__(self, data, columns=None):
+        if isinstance(data, dict):
+            self._columnNames = data.keys()
+            self._columns = [PurePythonSeries(val) for val in data.values()]
+        elif isinstance(data, list):
+            # note this for is not pandas-compliant:
```

Comment:
it means i was in a rush

On Tue, Nov 17, 2015 at 1:05 PM, Braxton McKee <notifications@github.com>
wrote:

> In packages/python/pyfora/typeConverters/Pandas.py
> <https://github.com/ufora/ufora/pull/128#discussion_r45095252>:
>
> > +
> > +class _DataframeFactory:
> > +    def __call__(self, data):
> > +        return PurePythonDataFrame(data)
> > +
> > +
> > +class PurePythonDataFrame:
> > +    """
> > +    TODO `columns` attr is an Index of column names in Pandas.
> > +    """
> > +    def __init__(self, data, columns=None):
> > +        if isinstance(data, dict):
> > +            self._columnNames = data.keys()
> > +            self._columns = [PurePythonSeries(val) for val in data.values()]
> > +        elif isinstance(data, list):
> > +            # note this for is not pandas-compliant:
>
> "this for" means what?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ufora/ufora/pull/128/files#r45095252>.
>



-- 
*Thomas Peters* | Software Engineer | *U**FORA *
115 West 27th Street 7th Floor, New York, NY 10001
Office: 646-918-6435 | Mobile: 732-610-8565
tpeters@ufora.com | www.ufora.com